src/Illuminate/Http/FileHelpers.php

```diff
@@ -46,6 +48,6 @@ public function hashName($path = null)
             $path = rtrim($path, '/').'/';
         }
 
-        return $path.md5_file($this->getRealPath()).'.'.$this->guessExtension();
+        return $path.Uuid::uuid4()->toString().'.'.$this->guessExtension();
```

Comment:
@themsaid this is a great new addition.
Nevertheless `md5` is great as it only contains hexadecimals, and allows to structure the filename splitting parts of its name:

`0123456789abcdef.jpg` => `01/23/45/67/89/ab/cd/ef.jpg`

But `UUID4` also contains `-`.

I believe this character could be removed for readability purpose?

```php
str_replace('-', '',  Uuid::uuid4()->toString())...
```